pkg/acsengine/pki.go

```diff
@@ -87,22 +92,74 @@ func CreatePki(extraFQDNs []string, extraIPs []net.IP, clusterDomain string, caP
 		errors <- err
 	}()
 
+	go func() {
+		var err error
+		organization := make([]string, 1)
+		organization[0] = "system:masters"
+		ip := net.ParseIP("127.0.0.1").To4()
+		peerIPs := append(extraIPs, ip)
+		etcdServerCertificate, etcdServerPrivateKey, err = createCertificate("etcd", caCertificate, caPrivateKey, false, nil, peerIPs, organization)
```

Comment:
Use different common names for etcd server/client/peer.

Should the isServer field be true?

Not sure whether extraIPs is needed here.

Reply:
ExtraIPs is needed because the certs need the peer IPs as SANs in order for tls to work. 

I think you're right that I should differentiate server/client. This was "hack" to get it to work by giving all the etcd certs clientAuth, servAuth and peer IPs. If you look at the if loop isServer needs to be false so that it goes to `else if CN == "etcd"` but this can probably be done more elegantly.